### PR TITLE
costa: support plugin install from npm package tarball

### PR DESCRIPTION
**Pull request by FeelyChau**
Legacy problems:
The downloaded file is saved to `constants.PIPCOOK_TMPDIR`, it will not be removed when install is finished, but it was managed by the OS.

### CODE REVIEW

**Comment by yorkie** [comment]
> The downloaded file is saved to os.tmpDir(), it will not be removed when install is finished, but it was managed by the OS.

We have the PIPCOOK_TMPDIR in core.

File: packages/costa/src/runtime.ts [hunk 1]
```diff
@@ -66,17 +71,25 @@ function fetchPackageJsonFromGit(remote: string, head: string): Promise<any> {
...
+    `--remote=${remote}`,
+    head,
+    'package.json'
+  ]);
+  return extractPackageJsonFromReadable(child.stdout, 'package.json');
+}
+
+function uncompressPackageJsonFromTgz(filename: string): Promise<any> {
+  const stream = fs.createReadStream(filename);
+  return extractPackageJsonFromReadable(stream.pipe(createUnzip()), 'package/package.json');
```
[the remark this answers is not in the document]
Comment (FeelyChau) [reply packages/costa/src/runtime.ts:90]: Hard code in npm/cli.
https://github.com/npm/cli/blob/3e7ed30d6e9211e39bd93ec4e254cc5a2b159947/lib/pack.js#L146
this PR supported the file generated from `npm pack`, should it support the custom format?

[the remark this answers is not in the document]
Comment (yorkie) [reply packages/costa/src/runtime.ts:90]: Ok, please add a comment for the magic thing.

File: packages/costa/src/runtime.ts [hunk 2]
```diff
@@ -48,12 +53,12 @@ function spawnAsync(command: string, args?: string[], opts: SpawnOptions = {}):
   });
 }
 
-function fetchPackageJsonFromGit(remote: string, head: string): Promise<any> {
+function extractPackageJsonFromReadable(readable: Readable, pkgFilename: string): Promise<any> {
```
[the remark this answers is not in the document]
Comment (FeelyChau) [reply packages/costa/src/runtime.ts:56]: The arg `readable` is not only a tar file, but also a Readable object from the file, I think `FromReadable` is more appropriate :).

[the remark this answers is not in the document]
Comment (yorkie) [reply packages/costa/src/runtime.ts:55]: Why not using Readable as the input of this function? I think the URL of tar is enough.

**Pull request merged**